oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java

```diff
@@ -141,6 +158,76 @@ private final File getWellKnownCredentialsFile() {
     return credentialFilePath;
   }
 
+  private boolean runningOnAppEngine() {
+    Class<?> systemPropertyClass = null;
+    try {
+      systemPropertyClass = forName(APP_ENGINE_SIGNAL_CLASS);
+    } catch (ClassNotFoundException expected) {
+      // SystemProperty will always be present on App Engine.
+      return false;
+    }
+    Exception cause = null;
+    Field environmentField;
+    try {
+      environmentField = systemPropertyClass.getField("environment");
+      Object environmentValue = environmentField.get(null);
+      Class<?> environmentType = environmentField.getType();
+      Method valueMethod = environmentType.getMethod("value");
+      Object environmentValueValue = valueMethod.invoke(environmentValue);
+      return (environmentValueValue != null);
+    } catch (NoSuchFieldException exception) {
```

Comment:
Instead of catching `NoSuchFieldException`, `IllegalAccessException`, `NoSuchMethodException`, and `InvocationTargetException`, would it be better to just catch `ReflectiveOperationException`?